edb/edgeql/parser/grammar/sdl.py

```diff
@@ -0,0 +1,975 @@
+
+
+# top-level SDL statements
+#
+class SDLStatement(Nonterm):
+    def reduce_IMPORT_ImportModuleList(self, *kids):
+        self.val = esast.Import(modules=kids[1].val)
+
+    def reduce_IMPORT_LPAREN_ImportModuleList_RPAREN(self, *kids):
+        self.val = esast.Import(modules=kids[2].val)
+
+    def reduce_ScalarTypeDeclaration(self, *kids):
+        self.val = kids[0].val
+
+    def reduce_AttributeDeclaration(self, *kids):
+        self.val = kids[0].val
+
+    def reduce_ObjectTypeDeclaration(self, *kids):
+        self.val = kids[0].val
+
+    def reduce_ViewDeclaration(self, *kids):
+        self.val = kids[0].val
+
+    def reduce_ConstraintDeclaration(self, *kids):
+        self.val = kids[0].val
+
+    def reduce_LinkDeclaration(self, *kids):
+        self.val = kids[0].val
+
+    def reduce_PropertyDeclaration(self, *kids):
+        self.val = kids[0].val
+
+    def reduce_FunctionDeclaration(self, *kids):
+        self.val = kids[0].val
+
+    def reduce_Index(self, *kids):
+        self.val = kids[0].val
+
+
+class DotName(Nonterm):
+    def reduce_ModuleName(self, *kids):
+        self.val = '.'.join(part for part in kids[0].val)
+
+
+class ImportModule(Nonterm):
+    def reduce_DotName(self, *kids):
+        self.val = esast.ImportModule(module=kids[0].val)
+
+    def reduce_DotName_AS_AnyIdentifier(self, *kids):
+        self.val = esast.ImportModule(module=kids[0].val,
+                                      alias=kids[2].val)
+
+
+class ImportModuleList(ListNonterm, element=ImportModule,
+                       separator=tokens.T_COMMA):
+    pass
+
+
+class Semicolons(Nonterm):
+    # one or more semicolons
+    def reduce_SEMICOLON(self, tok):
+        self.val = tok
+
+    def reduce_Semicolons_SEMICOLON(self, *kids):
+        self.val = kids[0].val
+
+
+class OptSemicolons(Nonterm):
+    def reduce_Semicolons(self, *kids):
+        self.val = kids[0].val
+
+    def reduce_empty(self):
+        self.val = None
+
+
+class SDLStatements(ListNonterm, element=SDLStatement,
+                    separator=Semicolons):
+    pass
+
+
+def _new_nonterm(clsname, clsdict={}, clskwds={}, clsbases=(Nonterm,)):
+    mod = sys.modules[__name__]
+
+    def clsexec(ns):
+        ns['__module__'] = __name__
+        for k, v in clsdict.items():
+            ns[k] = v
+        return ns
+
+    cls = types.new_class(clsname, clsbases, clskwds, clsexec)
+    setattr(mod, clsname, cls)
+    return cls
+
+
+class ProductionHelper:
+    def _passthrough(self, cmd):
+        self.val = cmd.val
+
+    def _singleton_list(self, cmd):
+        self.val = [cmd.val]
+
+    def _empty(self):
+        self.val = []
+
+    def _block(self, lbrace, cmdlist, sc2, rbrace):
+        self.val = cmdlist.val
+
+    def _block2(self, lbrace, sc1, cmdlist, sc2, rbrace):
+        self.val = cmdlist.val
+
+
+def commands_block(parent, *commands, opt=True):
+    if parent is None:
+        parent = ''
+
+    clsdict = collections.OrderedDict()
```

Comment:
No need to use OrderedDict since Python 3.6 (except for LRU caches)

Reply:
Dropped.